torchdata/dataloader2/communication/eventloop.py

```diff
@@ -26,16 +29,57 @@
 
 __all__ = [
     "DataPipeToQueuesLoop",
-    "SpawnProcessForDataPipeline",
-    "SpawnThreadForDataPipeline",
+    "CreateProcessForDataPipeline",
+    "CreateProcessForMultipleDataPipelines",
+    "CreateThreadForDataPipeline",
 ]
 
 
+def MultipleDataPipesToQueuesLoop(source_datapipes, req_queues, res_queues, call_on_process_init=None):
+    r"""
+    Set the appropriate pipes and protocol server type, and create a loop over multiple datapipes
+    with the protocol server in a non-blocking manner.
+    """
+    assert call_on_process_init is None, "``MultipleDataPipesToQueuesLoop`` does not support call_on_process_init"
+    assert len(source_datapipes) == len(req_queues) and len(req_queues) == len(
+        res_queues
+    ), "``MultipleDataPipesToQueuesLoop`` requires the same number of datapipes, request queues and response queues"
+
+    torch.set_num_threads(1)
+
+    loops = []
+
+    for source_datapipe, req_queue, res_queue in zip(source_datapipes, req_queues, res_queues):
+        loops.append(
+            _create_datapipe_queue_loop(source_datapipe, req_queue, res_queue, blocking_request_get=False)
+        )  # Non-blocking request
+
+    # Using `zip_longest` to guarantee the process is terminated only when
+    # all loops have received `TerminateRequest`
+    for _ in zip_longest(*loops):
+        # time.sleep to make Python switch context to get/send message in mp.Queue
+        # TODO(ejguan): Microbenchmarked a synthetic non-shardable case that sleep perform similar to pass.
+        #               A more comprehensive benchmarking in read-world scneario is needed.
```

Comment:
nit: real world? 😄

Reply:
Lol. I mean the OSS use case. I will reword it